fe4u-boilerplate-main/src/js/teachers/teacherList.js

```diff
@@ -0,0 +1,207 @@
+        if (teachers.length > 0) {
+            teachers.forEach((teacher) => {
+                const teacherCard = TeacherList.createTeacherCard(teacher);
+                teacherCard.onclick = () => this.onClickCardTeacher(teacher);
+                teacherList.appendChild(teacherCard);
+            });
+        }
+        return teacherList;
+    }
+
+    setStartTeacherList() {
+        if (this.teacherList) return;
+        this.teacherList = this.createTeacherList(this.teachers);
+        this.teacherList.firstChild && this.teacherGallery.appendChild(this.teacherList);
+        this.teachers.forEach((teacher) => {
+            this.teacherTableList.add(teacher);
+            if (teacher.favorite) {
+                const teacherLiElm = this.createFavTeacherCard(teacher);
+                this.teacherFavList.add(teacherLiElm);
+            }
+        });
+        this.teacherTableList.updateTable();
+    }
+
+    add(teacherDate) {
+        const teacher = Process.createUser(teacherDate, this.teachers.length + 1, true);
+        const teacherCard = TeacherList.createTeacherCard(teacher);
+        teacherCard.onclick = () => this.onClickCardTeacher(teacher);
+        this.teachers.push(teacher);
+        this.teacherList.appendChild(teacherCard);
+
+        this.teacherTableList.add(teacher);
+        this.teacherTableList.setupPage(this.teacherTableList.page.current);
+
+    };
+
+    activeMore(onClickMore) {
+        this.loadMoreBtn.addEventListener('click', async () => {
+            this.loadMoreBtn.classList.add('active');
+            onClickMore()
+                .then((res) => {
+                    this.loadMoreBtn.classList.remove('active');
+                    if (!res.ok) {
+                        this.loadMoreBtn.classList.add('not_work');
+                        this.loadMoreBtn.setAttribute("disabled", "");
+                    }
+                });
+        });
+    }
+
+    clear() {
+        if (this.teacherList && this.teacherGallery.firstChild === this.teacherList) {
+            this.teacherGallery.removeChild(this.teacherList);
+        } else if (this.teacherlistFiltered && this.teacherGallery.firstChild === this.teacherlistFiltered) {
+            this.teacherGallery.removeChild(this.teacherlistFiltered);
+        }
+    }
+
+    applySearchElements(opts = {}) {
+        if (!Object.keys(opts).length) {
+            this.resetFilterElements();
+            this.teacherlistFiltered = [];
+        } else {
+            const teachersSearched = SearchUser(this.teachers, opts);
+            console.log(teachersSearched);
+            this.teacherlistFiltered = this.createTeacherList(teachersSearched);
+            this.teacherGallery.innerHTML = ""
+            this.teacherGallery.appendChild(this.teacherlistFiltered);
+        }
+    }
+
+    setDataForChart() {
+        const TitleValue = Process.uniqueOption(this.teachers);
+        const Value = {};
+        const value = [];
```

Comment:
2 variables, which differ only with capitalized first letter, is difficult to distinguish. Please use other names.